### PR TITLE
BUG: frame.loc[2:, 'z'] not setting inplace when multi-block

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/frame/indexing/test_setitem.py [hunk 1]
```diff
@@ -407,6 +407,11 @@ def test_setitem_frame_duplicate_columns(self, using_array_manager):
             expected["C"] = expected["C"].astype("int64")
             # TODO(ArrayManager) .loc still overwrites
             expected["B"] = expected["B"].astype("int64")
+
+            mark = pytest.mark.xfail(
```
Comment (jreback) [inline pandas/tests/frame/indexing/test_setitem.py:411]: what this about?

**Comment by jreback** [comment]
is this user facing at all?

File: pandas/tests/frame/indexing/test_setitem.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/frame/indexing/test_setitem.py:411]: the ArrayManager behavior is different for reasons i'm still chasing down

**Comment by jbrockmendel** [comment]
> is this user facing at all?

yes, will update with whatsnew

**Comment by jbrockmendel** [comment]
whatsnew added + green

**Comment by jreback** [comment]
cc @phofl if you can have a look

**Review by phofl — APPROVED** [review]
lgtm

**Pull request merged**